modules/axp/Animal/lib/Racer.pm

```diff
@@ -0,0 +1,231 @@
+package Racer;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Carp qw( croak );
+use File::Slurp qw ( read_file write_file );
+use JSON;
+use Moose::Role;
+use Moose::Meta::Attribute::Native::Trait::Counter;
+use namespace::autoclean;
+
+=head1 NAME
+
+Racer - A generic thing that can generically race
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+=head1 SYNOPSIS
+
+Have you seen my fabulous racing snails? I use ladybirds as jockeys.
+
+=head1 SUBROUTINES/METHODS
+
+=head2 wins
+
+Number of wins
+
+=head2 won
+
+Racer won a race! :)
+
+=cut
+
+has 'wins' => (
+	traits => ['Counter'],
+	is => 'ro',
+	isa => 'Int',
+	default => 0,
+	handles => { won => 'inc' } );
```

Comment:
Behold the fabulous counter pragma! This creates both a `wins` getter and a `won` incrementor.